tenzir/tenzir.cpp

```diff
@@ -84,18 +84,22 @@ int main(int argc, char** argv) {
   const auto app_name = last_slash == std::string_view::npos
                           ? app_path
                           : app_path.substr(last_slash + 1);
-  if (app_name == "tenzir" || app_name == "tenzir-ctl") {
-    bool is_server = invocation->full_name == "start"
-                     || invocation->full_name == "exec"
-                     || caf::get_or(cfg.content, "tenzir.node", false);
-    std::string_view max_threads_key = "caf.scheduler.max-threads";
-    if (!is_server
-        && !caf::holds_alternative<caf::config_value::integer>(cfg,
-                                                               max_threads_key))
-      cfg.set(max_threads_key, 2);
-  }
+  bool is_server = false;
+  if (app_name == "tenzir-node")
+    is_server = true;
+  if (app_name == "vast" || app_name == "tenzir-ctl")
+    is_server = invocation->full_name == "start"
+                || caf::get_or(cfg.content, "tenzir.node", false);
+  else if (app_name == "tenzir")
+    is_server = invocation->full_name == "exec"
+                || caf::get_or(cfg.content, "tenzir.node", false);
```

Comment:
This can be shortened, and I believe `tenzir exec` does not create a node, so that condition was wrong.
```suggestion
    const bool is_server = app_name == "tenzir-node"
                         || caf::get_or(cfg.content, "tenzir.node", false)
                         || ((app_name == "vast" || app_name == "tenzir-ctl")
                             && invocation->full_name == "start");
```

Reply:
Makes sense, I slightly rearranged it to avoid the nested boolean expressions.